src/realm/object-store/shared_realm.cpp

```diff
@@ -133,7 +133,8 @@ std::shared_ptr<Transaction> Realm::transaction_ref()
 
 std::shared_ptr<Transaction> Realm::duplicate() const
 {
-    return m_coordinator->begin_read(read_transaction_version(), is_frozen());
+    auto version = read_transaction_version(); // does the validity check first
+    return m_coordinator->begin_read(version, is_frozen());
```

Comment:
I am not sure how we can end up into this situation where the Realm does not have a valid coordinator object (other than testing)

Reply:
This one comes from: realm-object-store-tests "SharedRealm: close()" -c "all functions throw ClosedRealmException after close"
```
    frame #5: 0x00000001016c8994 realm-object-store-tests`realm::Realm::duplicate(this=0x00000001078079f8) const at shared_realm.cpp:136:27 [opt]
    frame #6: 0x000000010098faac realm-object-store-tests`CATCH2_INTERNAL_TEST_165() at realm.cpp:2863:9 [opt]
(lldb) fr s 6
frame #6: 0x000000010098faac realm-object-store-tests`CATCH2_INTERNAL_TEST_165() at realm.cpp:2863:9 [opt]
   2860         REQUIRE_FALSE(realm->is_frozen());
   2861         REQUIRE_EXCEPTION(realm->get_number_of_versions(), ClosedRealm, msg);
   2862         REQUIRE_EXCEPTION(realm->read_transaction_version(), ClosedRealm, msg);
-> 2863         REQUIRE_EXCEPTION(realm->duplicate(), ClosedRealm, msg);
   2864
   2865         REQUIRE_EXCEPTION(realm->enable_wait_for_change(), ClosedRealm, msg);
   2866         REQUIRE_EXCEPTION(realm->wait_for_change(), ClosedRealm, msg);
(lldb) do
frame #5: 0x00000001016c8994 realm-object-store-tests`realm::Realm::duplicate(this=0x00000001078079f8) const at shared_realm.cpp:136:27 [opt]
   133
   134  std::shared_ptr<Transaction> Realm::duplicate() const
   135  {
-> 136      return m_coordinator->begin_read(read_transaction_version(), is_frozen());
   137  }
   138
   139  std::shared_ptr<DB>& Realm::Internal::get_db(Realm& realm)
```